Bottom: ef12221b9b980010ecf33567429f0b17dee2a194
Top:    010967cd7c516d8b9f92d5fd3c1056d94d2c9879
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-01 13:18:12 +0100

toporouter: Re-implement point_from_point_to_point() with atan2

Simplifies the code a lot


---

diff --git a/src/toporouter.c b/src/toporouter.c
index e5310b0..5a264ea 100644
--- a/src/toporouter.c
+++ b/src/toporouter.c
@@ -231,7 +231,7 @@ toporouter_arc_class(void)
   return klass;
 }
 
-#define MARGIN 10.0f
+#define MARGIN 10.0f /* XXX: Not sure - seems to be used to offset cairo drawing - thus could be in Coords */
 
 drawing_context_t *
 toporouter_output_init(int w, int h, char *filename) 
@@ -721,46 +721,10 @@ print_trace (void)
 void
 point_from_point_to_point(toporouter_vertex_t *a, toporouter_vertex_t *b, gdouble d, gdouble *x, gdouble *y)
 {
-  gdouble dx = vx(b) - vx(a);
-  gdouble dy = vy(b) - vy(a);
-  gdouble theta = atan(fabs(dy/dx));
-
-//#ifdef DEBUG_EXPORT  
-  if(!finite(theta)) {
-//    printf("!finte(theta): a = %f,%f b = %f,%f d = %f\n", vx(a), vy(a), vx(b), vy(b), d);
-//    print_trace();
-    //TODO: this shouldn't happen, fix the hack
-    *x = vx(a);
-    *y = vy(a);
-    return;
-  }
-//#endif
-
-  g_assert(finite(theta));
-
-  *x = vx(a); *y = vy(a);
-
-  if( dx >= 0. ) {
+  double theta = atan2 (vy(b) - vy(a), vx(b) - vx(a));
 
-    if( dy >= 0. ) {
-      *x += d * cos(theta);
-      *y += d * sin(theta);
-    }else{
-      *x += d * cos(theta);
-      *y -= d * sin(theta);
-    }
-
-  }else{
-    
-    if( dy >= 0. ) {
-      *x -= d * cos(theta);
-      *y += d * sin(theta);
-    }else{
-      *x -= d * cos(theta);
-      *y -= d * sin(theta);
-    }
-
-  }
+  *x = vx(a) + d * cos (theta);
+  *y = vy(a) + d * sin (theta);
 }
